crates/primitives/starknet/src/fees/mod.rs

```diff
@@ -0,0 +1,251 @@
+    account_tx_context: &AccountTransactionContext,
+    actual_fee: Fee,
+) -> Result<CallInfo, TransactionExecutionErrorWrapper> {
+    // TODO: use real value.
+    // FIXME: https://github.com/keep-starknet-strange/madara/issues/331
+    let max_fee = Fee(u128::MAX);
+    if actual_fee > max_fee {
+        return Err(TransactionExecutionErrorWrapper::FeeTransferError { max_fee, actual_fee });
+    }
+    // TODO: This is what's done in the blockifier but this should be improved.
+    // FIXME: https://github.com/keep-starknet-strange/madara/issues/332
+    // The least significant 128 bits of the amount transferred.
+    let lsb_amount = StarkFelt::from(actual_fee.0 as u64);
+    // The most significant 128 bits of the amount transferred.
+    let msb_amount = StarkFelt::from(0);
+
+    let storage_address = block_context.fee_token_address;
+    let fee_transfer_call = CallEntryPoint {
+        class_hash: None,
+        code_address: None,
+        entry_point_type: EntryPointType::External,
+        entry_point_selector: EntryPointSelector(
+            // The value is hardcoded and it's the encoding of the "transfer" selector so it cannot fail.
+            StarkFelt::new([
+                0, 131, 175, 211, 244, 202, 237, 198, 238, 191, 68, 36, 111, 229, 78, 56, 201, 94, 49, 121, 165, 236,
+                158, 168, 23, 64, 236, 165, 180, 130, 209, 46,
+            ])
+            .unwrap(),
+        ),
+        calldata: calldata![
+            *block_context.sequencer_address.0.key(), // Recipient.
+            lsb_amount,
+            msb_amount
+        ],
+        storage_address,
+        caller_address: account_tx_context.sender_address,
+        call_type: CallType::Call,
+    };
+    let mut execution_context = ExecutionContext::default();
+
+    fee_transfer_call
+        .execute(state, execution_resources, &mut execution_context, block_context, account_tx_context)
+        .map_err(TransactionExecutionErrorWrapper::EntrypointExecution)
+}
+
+/// Computes the fees from the execution resources.
+///
+/// # Arguments
+///
+/// * `resources` - Execution resources to compute the fees from.
+/// * `block_context` - Block context to get information needed to compute the fees.
+///
+/// # Returns
+///
+/// [Fee] - the fees computed for the transaction.
+///
+/// # Error
+///
+/// [TransactionExecutionErrorWrapper] - if the computation of the l1 gas usage fails, returns an
+/// error.
+pub fn calculate_tx_fee(
+    resources: &BTreeMap<String, usize>,
+    block_context: &BlockContext,
+) -> Result<Fee, TransactionExecutionErrorWrapper> {
+    let (l1_gas_usage, vm_resources) = extract_l1_gas_and_vm_usage(resources);
+    let l1_gas_by_vm_usage = calculate_l1_gas_by_vm_usage(block_context, &vm_resources)?;
+    let total_l1_gas_usage = l1_gas_usage as f64 + l1_gas_by_vm_usage;
+    // Ceil is in the std lib so we can't use it sadly.
+    let total_l1_gas_usage = if total_l1_gas_usage - total_l1_gas_usage as u128 as f64 > 0.0 {
```

Comment:
We could use `libm` or `micromath` crates here

Reply:
does it bring any value to add a new dep ?